docs/apis/create-with-equality-fn.md

```diff
@@ -0,0 +1,618 @@
+  const position = usePositionStore()
+
+  return (
+    <div
+      style={{
+        position: 'relative',
+        width: '100vw',
+        height: '100vh',
+      }}
+    >
+      <div
+        style={{
+          position: 'absolute',
+          backgroundColor: 'red',
+          borderRadius: '50%',
+          transform: `translate(${position.x}px, ${position.y}px)`,
+          left: -10,
+          top: -10,
+          width: 20,
+          height: 20,
+        }}
+        onMouseEnter={(event) => {
+          const parent = event.currentTarget.parentElement
+          const parentWidth = parent.clientWidth
+          const parentHeight = parent.clientHeight
+
+          setPosition({
+            x: Math.ceil(Math.random() * parentWidth),
+            y: Math.ceil(Math.random() * parentHeight),
+          })
+        }}
+      />
+    </div>
+  )
+}
+```
+
+### Subscribing to state updates
+
+By subscribing to state updates, you register a callback that fires whenever the store's state
+updates. We can use `subscribe` for external state management.
+
+```tsx
+import { useEffect } from 'react'
+import { createWithEqualityFn } from 'zustand/traditional'
+import { shallow } from 'zustand/vanilla/shallow'
+
+type PositionStoreState = { x: number; y: number }
+
+type PositionStoreActions = {
+  setPosition: (nextPosition: Partial<PositionStoreState>) => void
+}
+
+type PositionStore = PositionStoreState & PositionStoreActions
+
+const usePositionStore = createWithEqualityFn<PositionStore>()(
+  (set) => ({
+    x: 0,
+    y: 0,
+    setPosition: (nextPosition) => {
+      set(nextPosition)
+    },
+  }),
+  shallow,
+)
+
+export default function MovingDot() {
+  const [position, setPosition] = usePositionStore((state) => [
+    { x: state.x, y: state.y },
+    state.setPosition,
+  ])
+
+  useEffect(() => {
+    const unsubscribePositionStore = usePositionStore.subscribe(({ x, y }) => {
+      console.log('new position', { position: { x, y } })
+    })
+
+    return () => {
+      unsubscribePositionStore()
+    }
+  }, [])
+
+  return (
+    <div
+      style={{
+        position: 'relative',
+        width: '100vw',
+        height: '100vh',
+      }}
+    >
+      <div
+        style={{
+          position: 'absolute',
+          backgroundColor: 'red',
+          borderRadius: '50%',
+          transform: `translate(${position.x}px, ${position.y}px)`,
+          left: -10,
+          top: -10,
+          width: 20,
+          height: 20,
+        }}
+        onMouseEnter={(event) => {
+          const parent = event.currentTarget.parentElement
+          const parentWidth = parent.clientWidth
+          const parentHeight = parent.clientHeight
+
+          setPosition({
+            x: Math.ceil(Math.random() * parentWidth),
+            y: Math.ceil(Math.random() * parentHeight),
+          })
+        }}
+      />
+    </div>
+  )
+}
+```
+
+## Troubleshooting
+
+### I’ve updated the state, but the screen doesn’t update
```

Comment:
You can link the "you should always use immutable updates" note to here to visualize what happens if you don't.

Reply:
@sewera nice suggestion